.github/workflows/e2e-typescript-workflow.yml

```diff
@@ -47,6 +47,9 @@ jobs:
         yarn init -p
         yarn add typescript
 
+        cp "$HERE_DIR"/scripts/test-typescript-patch.js .
+        yarn node ./test-typescript-patch.js
```

Comment:
Also worth noting: I've added a few tests to prevent regressions on the module import logic.